### DISCUSSION TITLE
Filtrowanie wg kraju, nadajnika, nazwy, daty pierwszego odbioru

### DISCUSSION THREAD

**Discussion by HyperDX** [TomaszGasior/RadioLista-v3#145] [2. Pomysły Ideas]
Myślę, że bardzo fajna funkcja, polegałaby ona na tym, że (zakładając, że użytkownik tak samo nazwał te same stacje, nadajniki oraz kraje)klikając na nazwę kraju np. Niemcy, D, DE (zależy jak kto wpisał) to wyświetla nam zamiast całego wykazu stacje które w polu kraj mają właśnie tak wpisane. Tak samo z nazwą stacji, nadajnikiem i datą pierwszego odbioru. Sądzę, że bardzo fajnie rozwiązanie, ale nie potrzebne od teraz :).

Zauważyłem też, że przez właśnie sortowanie stacji radiowych może być problem z ich edytowaniem. Może teraz edytowanie w związku z tym przeniesie się na pole z częstotliwością?

**Comment by TomaszGasior** [comment]
Samo w sobie filtrowanie — bo tak trzeba by nazwać tę funkcję — byłoby na pewno ciekawą opcją, zwłaszcza w wykazach z 200 i więcej stacji. 

Implementacja samego filtrowania raczej nie byłaby trudna, natomiast zagwozdką jest UI. Jak to mądrze umieścić w interfejsie?
- Można zrealizować UI dokładnie tak, jak opisano wyżej — przez klikalne linki w tabeli wykazu sortujące wg klikniętej wartości. Nie dośc, że trzeba by wtedy zmienić sposób edycji stacji, to nie jestem przekonany do intuicyjności tego rozwiązania.
- Inna opcja to inspirowanie się arkuszami kalkulacyjnymi Excel/Calc, gdzie obok nazwy filtrowalnej kolumny jest np. ikona lejka, po której kliknięciu pojawia się popup umożliwiający przeprowadzenie lub wycofanie filtru. Tutaj implementacja takiego UI więcej by zajęła.

Spośród wszystkich obecnych kolumn wykazu poniższe są sortowalne:
- DISTANCE
- FREQUENCY
- POWER
- PRIVATE_NUMBER
- QUALITY
- COUNTRY
- FIRST_LOG_DATE
- LOCALITY
- LOCATION
- NAME
- POLARIZATION
- RADIO_GROUP
- RDS_PI
- TYPE

Jakie kolumny byłyby filtrowalne? Zapewne dokładnie te same za wyjątkiem liczbowych, ale z jakością odbioru.

Odkładam tę funkcję na później: po nowych kolumnach, dostępności dla czytników ekranowych i statystykach.
